### PR TITLE
Resolve symlinks on directory listing

### PR DESCRIPTION
**Pull request by aliemjay**
This has the benefit of showing the size and modification date of the pointed-to file. Symlink to directories now respects '--dirs-first' option and broken symlinks don't show up in directory listing.

The only downside AFAIK is more syscalls when listing directories.

@svenstaro Any input on this? should it be configurable? can we abandon the symlink symbol?

### CODE REVIEW

**Comment by svenstaro** [comment]
Could you rebase this?

**Comment by aliemjay** [comment]
@svenstaro done.

**Comment by svenstaro** [comment]
I quite like this as it is. I like keeping the symlink symbol around. Of course, I'd appreciate a test for this if you want to make one but I'd say for this change it's optional. Once you're happy, remove the `[RFC]` prefix and I'll merge it. :)

**Comment by aliemjay** [comment]
RFC tag removed!

CSS styles may still need some cleanup though: `symlink` class is no longer used, `symlink-symbol` border color seems off to me.

**Comment by svenstaro** [comment]
Alright, feel free to remove the disused class, no reason to keep it around unless we maybe want to make use of it still? Perhaps we can underline the symlinks in the symlink color?

File: tests/fixtures/mod.rs [hunk 1]
```diff
@@ -42,23 +48,42 @@ pub fn tmpdir() -> TempDir {
     let tmpdir = assert_fs::TempDir::new().expect("Couldn't create a temp dir for tests");
     let mut files = FILES.to_vec();
     files.extend_from_slice(HIDDEN_FILES);
-    for file in &files {
+    let (files_hard, files_symlink) = files.split_at(1);
+    println!("path = {:?}", tmpdir.path());
```
Comment (svenstaro) [inline tests/fixtures/mod.rs:52]: Is this a debug leftover?

File: tests/fixtures/mod.rs [hunk 2]
```diff
@@ -42,23 +48,42 @@ pub fn tmpdir() -> TempDir {
...
+    for file in files_hard {
         tmpdir
             .child(file)
             .write_str("Test Hello Yes")
             .expect("Couldn't write to file");
     }
+    for file in files_symlink {
```
Comment (svenstaro) [inline tests/fixtures/mod.rs:59]: Could you sprinkle a few comments in here about what we're doing in all the loops and such now that this fixture has become a bit more complex?

**Comment by svenstaro** [comment]
Cool stuff with the fixtures! Now perhaps just make one test check whether the symlinks are actually rendered as they should be (while other files and dirs are still rendered as _they_ should be.

File: tests/fixtures/mod.rs (hunk 1 above)
Comment (svenstaro) [reply tests/fixtures/mod.rs:52]: By the way, you can use `dbg!()` instead as an easier way to achieve exactly the same print!

**Comment by aliemjay** [comment]
> Cool stuff with the fixtures! Now perhaps just make one test check whether the symlinks are actually rendered as they should be (while other files and dirs are still rendered as _they_ should be.

I believe that the current tests would still fail with the old behaviour because symlink filenames do always have a trailing `/`.
But I think that a single standalone test is better and that symlink files and directories should be made local to such test. Fixing that soon..

**Comment by aliemjay** [comment]
I don't really have a strong opinion on symlink symbol style, but it looks ok to me without a special style.

**Comment by svenstaro** [comment]
@aliemjay could you resolve the conflict? I think this was in pretty good shape already the last time I looked at it.

**Comment by svenstaro** [comment]
Looking at this again, I'm thinking: Wouldn't it be kind of cool to render the pointed-to relative location of the symlink behind the arrow?

**Comment by aliemjay** [comment]
I have no strong opinion on this, but a symlink may be absolute or may point outside the serve_path  and that would show potentially unwanted path names.

**Comment by svenstaro** [comment]
Perhaps that's something we could show actually in case it's pointing to somewhere that would be off-limits?

**Comment by aliemjay** [comment]
Well, as I said,I have no strong opinion but I see that symlinks should be totally abstracted away from the user and should be used by the admin only to organize the file tree.

**Comment by svenstaro** [comment]
That's a fair point! How about a follow-up PR with a new flag to show symlink destination? :)

**Comment by aliemjay** [comment]
That would be interesting. I shall give it a try!

**Comment by svenstaro** [comment]
Merging this one as is! Follow ups: https://github.com/svenstaro/miniserve/issues/498 and https://github.com/svenstaro/miniserve/issues/499.

**Pull request merged**